server/mlabns/util/fqdn_rewrite.py

```diff
@@ -0,0 +1,102 @@
+"""Rewrites raw M-Lab FQDNs to apply post-processing or annotations."""
+
+import logging
+
+from mlabns.util import message
+
+
+def rewrite(fqdn, address_family, tool_id):
+    """Rewrites an FQDN to add necessary annotations and special-casing.
+
+    Performs the following rewrites on an FQDN:
+    * Adds a v4/v6 annotation if the client requested an explicit address family
+    * Applies a workaround to fix FQDNs for NDT-SSL queries.
+
+    Args:
+        fqdn: A tool FQDN with no address family specific annotation.
+        address_family: The address family for which to create the FQDN or None
+            to create an address family agnostic FQDN.
+        tool_id: Name of tool associated with the FQDN (e.g. 'ndt_ssl').
+
+    Returns:
+        FQDN after rewriting to apply all modifications to the raw FQDN.
+    """
+    rewritten_fqdn = _apply_af_awareness(fqdn, address_family)
```

Comment:
Sorry, I may not have explained myself well enough.  Let's say that someone queries mlab-ns using the ndt_ssl tool id:

http://mlab-ns.appspot.com/ndt_ssl?address_family=ipv4

... and is, say, returned this rewritten host name:

ndt-iupui-mlab2v4-mia01.measurement-lab.org

Name resolution will fail.  "flattened", AF decorated DNS RRs do not currently exist.  My comment here was just to make sure that if someone is querying for ndt_ssl that the fqdn is not passed through _apply_af_awareness().